x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/NoopStep.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.index.Index;
+
+/**
+ * This is a Noop step that can be used for backwards compatibility when removing some step in newer versions.
+ * It literally does nothing so that we can safely proceed to the nextStepKey without getting stuck.
+ */
+public class NoopStep extends ClusterStateWaitStep {
+    public static final String NAME = "NOOP";
```

Comment:
All of our step names are lowercased

```suggestion
    public static final String NAME = "noop";
```

Reply:
@dakrone thanks! I applied to suggestion so now the NAME is in lowercase